src/settings/OrderNumberForm.js

```diff
@@ -0,0 +1,159 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { Field } from 'redux-form';
+
+import {
+  sortBy,
+  uniqBy,
+} from 'lodash';
+
+import stripesForm from '@folio/stripes/form';
+import {
+  Accordion,
+  AccordionSet,
+  Button,
+  Checkbox,
+  Col,
+  Headline,
+  MultiSelection,
+  Pane,
+  Row,
+} from '@folio/stripes/components';
+
+class OrderNumberForm extends Component {
+  static propTypes = {
+    label: PropTypes.object.isRequired,
+    handleSubmit: PropTypes.func.isRequired,
+    pristine: PropTypes.bool,
+    submitting: PropTypes.bool,
+    initialValues: PropTypes.object,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      prefix: [],
+      suffix: [],
+    };
+  }
+
+  getLastMenu = () => {
+    const { pristine, submitting } = this.props;
+
+    return (
+      <Button
+        type="submit"
+        buttonStyle="primary"
+        disabled={(pristine || submitting)}
+        marginBottom0
+      >
+        <FormattedMessage id="ui-orders.settings.poNumber.saveBtn" />
+      </Button>
+    );
+  };
+
+  addPrefix = ({ inputValue }) => {
+    // eslint-disable-next-line react/no-access-state-in-setstate
```

Comment:
Actually, as a developer you're not supposed to silent lint rules, it's their purpose to check such situations for a reason. 
In this very case I think there is no reason to make the silent and it's OK to use `setState((oldSate) => newState)`